addons/l10n_in/tests/test_partner_details_on_invoice.py

```diff
@@ -175,3 +175,42 @@ def test_place_of_supply(self):
                 'l10n_in_state_id': self.partner_b.state_id.id,
             }]
         )
+
+    def test_l10n_in_hsn_summary_with_discount(self):
+        """ Line with discount """
+        move = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    'product_id': self.product_a.id,
+                    'quantity': 1.0,
+                    'price_unit': 100.0,
+                    'discount': 10.0,
+                    'tax_ids': [(6, 0, self.gst_18.ids)],
+                }),
+            ],
+        })
+        self.assertEqual(
+            move._l10n_in_get_hsn_summary_table(),
```

Comment:
If you add the `discount` parameter, you can move this test with the others tests about hsn_summary